### ISSUE TITLE
RPM build is failing on the rpath in the new shared libs

### ISSUE THREAD

**Issue by trink** [mozilla-services/heka#1311]
```
CPack: Create package
CPackRPM: Will use GENERATED spec file: /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/SPECS/heka.spec
CPackRPM:Debug: You may consult rpmbuild logs in:
CPackRPM:Debug:    - /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/rpmbuild.err
CPackRPM:Debug: *** + umask 022
+ cd /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/BUILD
+ mv /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/tmpBBroot
+ exit 0
+ umask 022
+ cd /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/BUILD
+ '[' /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 '!=' / ']'
+ rm -rf /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
++ dirname /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ mkdir -p /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM
+ mkdir /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ '[' -e /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 ']'
+ rm -rf /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ mv /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/tmpBBroot /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ /usr/lib/rpm/check-rpaths
*******************************************************************************
*
* WARNING: 'check-rpaths' detected a broken RPATH and will cause 'rpmbuild'
*          to fail. To ignore these errors, you can set the '$QA_RPATHS'
*          environment variable which is a bitmask allowing the values
*          below. The current value of QA_RPATHS is 0x0000.
*
*    0x0001 ... standard RPATHs (e.g. /usr/lib); such RPATHs are a minor
*               issue but are introducing redundant searchpaths without
*               providing a benefit. They can also cause errors in multilib
*               environments.
*    0x0002 ... invalid RPATHs; these are RPATHs which are neither absolute
*               nor relative filenames and can therefore be a SECURITY risk
*    0x0004 ... insecure RPATHs; these are relative RPATHs which are a
*               SECURITY risk
*    0x0008 ... the special '$ORIGIN' RPATHs are appearing after other
*               RPATHs; this is just a minor issue but usually unwanted
*    0x0010 ... the RPATH is empty; there is no reason for such RPATHs
*               and they cause unneeded work while loading libraries
*    0x0020 ... an RPATH references '..' of an absolute path; this will break
*               the functionality when the path before '..' is a symlink
*
*
* Examples:
* - to ignore standard and empty RPATHs, execute 'rpmbuild' like
*   $ QA_RPATHS=$[ 0x0001|0x0010 ] rpmbuild my-package.src.rpm
* - to check existing files, set $RPM_BUILD_ROOT and execute check-rpaths like
*   $ RPM_BUILD_ROOT=<top-dir> /usr/lib/rpm/check-rpaths
*
*******************************************************************************
ERROR   0002: file '/usr/share/heka/lua_modules/cjson.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/circular_buffer.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_b
ase/Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/circular_buffer.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Buil
d/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/lpeg.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/hyperloglog.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/
Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/hyperloglog.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lu
a_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/struct.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/bloom_filter.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base
/Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/bloom_filter.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/l
ua_sandbox/src]
error: Bad exit status from /var/tmp/rpm-tmp.eWd5zi (%install)
    Bad exit status from /var/tmp/rpm-tmp.eWd5zi (%install)
 ***
CPackRPM:Debug:    - /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/rpmbuild.out
CPackRPM:Debug: *** + umask 022
+ cd /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/BUILD
+ mv /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/tmpBBroot
+ exit 0
+ umask 022
+ cd /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/BUILD
+ '[' /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 '!=' / ']'
+ rm -rf /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
++ dirname /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ mkdir -p /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM
+ mkdir /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ '[' -e /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64 ']'
+ rm -rf /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ mv /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/tmpBBroot /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64
+ /usr/lib/rpm/check-rpaths
*******************************************************************************
*
* WARNING: 'check-rpaths' detected a broken RPATH and will cause 'rpmbuild'
*          to fail. To ignore these errors, you can set the '$QA_RPATHS'
*          environment variable which is a bitmask allowing the values
*          below. The current value of QA_RPATHS is 0x0000.
*
*    0x0001 ... standard RPATHs (e.g. /usr/lib); such RPATHs are a minor
*               issue but are introducing redundant searchpaths without
*               providing a benefit. They can also cause errors in multilib
*               environments.
*    0x0002 ... invalid RPATHs; these are RPATHs which are neither absolute
*               nor relative filenames and can therefore be a SECURITY risk
*    0x0004 ... insecure RPATHs; these are relative RPATHs which are a
*               SECURITY risk
*    0x0008 ... the special '$ORIGIN' RPATHs are appearing after other
*               RPATHs; this is just a minor issue but usually unwanted
*    0x0010 ... the RPATH is empty; there is no reason for such RPATHs
*               and they cause unneeded work while loading libraries
*    0x0020 ... an RPATH references '..' of an absolute path; this will break
*               the functionality when the path before '..' is a symlink
*
*
* Examples:
* - to ignore standard and empty RPATHs, execute 'rpmbuild' like
*   $ QA_RPATHS=$[ 0x0001|0x0010 ] rpmbuild my-package.src.rpm
* - to check existing files, set $RPM_BUILD_ROOT and execute check-rpaths like
*   $ RPM_BUILD_ROOT=<top-dir> /usr/lib/rpm/check-rpaths
*
*******************************************************************************
ERROR   0002: file '/usr/share/heka/lua_modules/cjson.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/circular_buffer.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_b
ase/Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/circular_buffer.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Buil
d/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/lpeg.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/hyperloglog.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/
Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/hyperloglog.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lu
a_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/struct.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib]
ERROR   0002: file '/usr/share/heka/lua_modules/bloom_filter.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base
/Build/lua_sandbox/src]
ERROR   0002: file '/usr/share/heka/lua_modules/bloom_filter.so' contains an invalid rpath '/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/src' in [$ORIGIN/../lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/lua_sandbox/ep_base/lib:/home/build/data-pipeline/build/heka/build/ep_base/Build/l
ua_sandbox/src]
error: Bad exit status from /var/tmp/rpm-tmp.eWd5zi (%install)
    Bad exit status from /var/tmp/rpm-tmp.eWd5zi (%install)
 ***
CPack Error: Problem copying the package: /home/build/data-pipeline/build/heka/build/_CPack_Packages/Linux/RPM/heka-0_9_0-linux-amd64.rpm to /home/build/data-pipeline/build/heka/build/heka-0_9_0-linux-amd64.rpm
CPack Error: Error when generating package: heka
Makefile:107: recipe for target 'package' failed
make: *** [package] Error 1
```
@whd

**Comment by trink** [comment]
The build rpath needs to be removed before packaging.

**Comment by trink** [comment]
It is an error to have an RPATH in the libraries or executable in an RPM. So the RPATHs will be removed. What this means:
1) Heka will only run when installed into the standard system directories
2) Only one version of Heka can be installed on a system at a time
3) There can only be one liblua.so (the lua_sandbox version) installed on the system.  Fortunately this will not conflict with the standard Lua RPM which installs liblua-5.1.so

**Comment by trink** [comment]
The only fatal failures were the two dependency failures on the RPM installation.  These have been resolved and the RPATHs remain unchanged so the tarball can still be installed to whatever location is desired and function properly.  At this point in time we will not address the other rpmlint errors and warnings.

**Comment by trink** [comment]
Closed via #1330
